### PR TITLE
Added Vaccination Records Table and Status Update Functionality 

### PR DESCRIPTION
**Pull request by AviraLS22**
*** Summary
This pull request implements the feature to manage vaccination records in the pet's edit view. Users can now view and modify vaccination statuses for each pet.

***Changes Made
- Added a table to display vaccination records with the following columns:
  - **Vaccine**: Name of the vaccine (e.g., Parvovirus, Deworming).
  - **Status**: A dropdown menu for selecting the vaccination status with options "PENDING" and "APPLIED".
  
- Replaced the previous Vaccinated checkbox with the new vaccination records table.
  
- Integrated functionality to update the vaccination status, which also updates the date to the current date (`LocalDate.now()`).

*** Acceptance Criteria
- Users can see the vaccination records displayed as a table.
- Users can select a vaccination status from a dropdown menu for each record.
- The status updates correctly and reflects the current date when modified.
- The vaccination records table appears instead of the Vaccinated checkbox.

*** Additional Notes
Please review the changes and provide feedback. Let me know if any further modifications are required!

### CODE REVIEW

**Comment by yashudhlani** [comment]
Hi @AviraLS22 , seems like the new branch you have created is from the previous branch and not from the main. 

I would suggest to follow this order before starting the code:
1. Checkout to main branch, rebase it from the origin
2. checkout to a new branch, then do the code changes.

for this, you can cherry pick the latest commit and create a new branch with it and then raise the PR.

Let me know if you have any more questions. Happy to help!

File: src/main/resources/templates/adoption/descriptionForAdoption.html [hunk 1]
```diff
@@ -16,7 +16,8 @@
                 <img style="width:320px;height:240px;" th:attr="src=@{${gcpImageUrl} + ${pet.images.get(0).uuid}}"/>
                 <div class="caption" align="center">
                     <h3 th:text="${pet.name}"></h3>
-                    <p th:text="${pet.birthDate}"/>
+                    <!-- Using custom dateFormatter to format birthDate -->
+                    <li th:text="${@dateFormatter.formatToDate(pet.birthDate, #locale)}"/>
```
Comment (josdem) [inline src/main/resources/templates/adoption/descriptionForAdoption.html:20]: Now looks good, thanks!

File: src/main/resources/templates/pet/edit.html [hunk 2]
```diff
@@ -1,128 +1,168 @@
...
+                                <table class="table">
+                                    <thead>
+                                        <tr>
+                                            <th>Vaccine</th>
+                                            <th>Status</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        <!-- Loop through the pet's vaccination records -->
+                                        <tr th:each="vaccine : ${petCommand.vaccinationRecords}">
```
Comment (josdem) [inline src/main/resources/templates/pet/edit.html:107]: Seems like `vaccinationRecords` do not exist in `PetCommand`, you might need to add and populate.
Also validate if the list is empty.

File: src/main/resources/templates/pet/edit.html [hunk 3]
```diff
@@ -1,128 +1,168 @@
...
+    <title th:text="#{pet.view.create.title}" />
+
+    <head th:insert="~{fragments/include}" />
     <script th:src="@{/assets/servizi-dog-theme/node_modules/jquery/dist/jquery.min.js}"></script>
     <script th:src="@{/assets/servizi-dog-theme/node_modules/bootstrap/dist/js/bootstrap.min.js}"></script>
     <script th:src="@{/javascript/selector.js}"></script>
 </head>
+
 <body class="login">
-<div th:insert="~{fragments/header}"/>
```
Comment (josdem) [inline src/main/resources/templates/pet/edit.html:12]: Please do not format this file, to make easier to see your changes.

File: src/main/resources/templates/user/create.html [hunk 4]
```diff
@@ -96,7 +98,6 @@
                                     <option value="+48">+48 (Poland)</option>
                                     <option value="+65">+65 (Singapore)</option>
                                     <option value="+60">+60 (Malaysia)</option>
-                                    <option value="+34">+34 (Spain)</option>
                                     <option value="+356">+356 (Malta)</option>
```
Comment (josdem) [inline src/main/resources/templates/user/create.html:101]: Any reason to delete Spain from country codes?

File: src/main/resources/templates/user/create.html [hunk 5]
```diff
@@ -105,7 +106,6 @@
                                     <option value="+98">+98 (Iran)</option>
                                     <option value="+972">+972 (Israel)</option>
                                     <option value="+90">+90 (Turkey)</option>
-                                    <option value="+62">+62 (Indonesia)</option>
                                     <option value="+353">+353 (Ireland)</option>
```
Comment (josdem) [inline src/main/resources/templates/user/create.html:109]: And Indonesia ?

File: src/main/resources/templates/user/create.html [hunk 6]
```diff
@@ -116,12 +116,13 @@
                                     <option value="+373">+373 (Moldova)</option>
                                     <!-- Add more country codes as needed -->
                                 </select>
-                                <input type="text" name="mobile" th:field="*{mobile}" class="form-control mobile-number"
-                                       placeholder="phone number" id="mobile"
-                                       pattern="\d{10}" title="Phone number must be 10 digits only" required />
+                                <input type="text" name="phone" th:field="*{phone}" class="form-control mobile-number"
```
Comment (josdem) [inline src/main/resources/templates/user/create.html:119]: The field name is `mobile`, please take a look at the class `UserCommand`, please revert this change.

**Review by josdem — CHANGES_REQUESTED** [review]
This time you have several comments, please take a look when you have a chance.

**Comment by josdem** [comment]
@all-contributors please add @yashudhlani  for mentoring

File: src/main/resources/templates/pet/giveForAdoption.html [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+';lbv <!DOCTYPE html>
 <html xmlns:th="https://www.thymeleaf.org" xmlns="http://www.w3.org/1999/html">
```
Comment (kaminuma) [inline src/main/resources/templates/pet/giveForAdoption.html:2]: 【typo】
The string [';lbv] before <!DOCTYPE html> seems to be a typo.
Having unnecessary string before <!DOCTYPE html> may cause an error.

**Review by kaminuma — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/user/create.html (hunk 4 above)
Comment (AviraLS22) [reply src/main/resources/templates/user/create.html:101]: NO sir , it was a mistake to remove Spain and Indonesia from the country codes.

**Comment by AviraLS22** [comment]
Thank you for the guidance, @yashudhlani . I will rebase my changes from the main branch and create a new branch to ensure everything is up to date.

**Comment by AviraLS22** [comment]
Thank you all for the valuable feedback!

@yashudhlani, I appreciate the suggestion regarding the branching strategy. I will rebase my branch from the main branch and create a new one to ensure everything is current.

@josdem, I will add the vaccinationRecords to the PetCommand class and validate the list as you mentioned. I will also revert the change related to the field name to mobile and be mindful of formatting in future commits.

@kaminuma, I’ll correct the typo in the HTML file.

Thanks again for your support! I'll address these changes and push an update soon.

File: src/main/resources/templates/user/create.html (hunk 4 above)
Comment (josdem) [reply src/main/resources/templates/user/create.html:101]: > NO sir , it was a mistake to remove Spain and Indonesia from the country codes.

Thanks for confirming

File: src/main/java/com/josdem/vetlog/model/Pet.java [hunk 8]
```diff
@@ -14,84 +14,96 @@
...
+     @JoinColumn(name = "pet_id")
+     private List<VaccinationRecord> vaccinationRecords;
+ 
+     @OneToMany(fetch = FetchType.LAZY)
+     @JoinColumn(name = "pet_image_id")
+     private List<PetImage> images;
+ 
+     // New Methods for Vaccination Records
+ 
+     public void setVaccinationRecords(List<VaccinationRecord> vaccinationRecords) {
```
Comment (yashudhlani) [inline src/main/java/com/josdem/vetlog/model/Pet.java:101]: @AviraLS22 , I don't know why the whole file change is getting shown here. 

But when you use getter and setter anotation, you don't need to explicitly add these methods.

**Review by yashudhlani — CHANGES_REQUESTED** [review]

File: src/main/java/com/josdem/vetlog/model/Pet.java (hunk 8 above)
Comment (AviraLS22) [reply src/main/java/com/josdem/vetlog/model/Pet.java]: @yashudhlani , I’ve removed the manually added getter and setter methods for vaccinationRecords as per the feedback. Lombok's @Getter and @Setter annotations are now used instead.

**Comment by AviraLS22** [comment]
Screenshot from 2024-09-29 22-10-19
Hi all,

I'm encountering a build failure in the CI pipeline related to starting MySQL. The error message is:

ERROR 1045 (28000): Access denied for user '-p'@'localhost' (using password: NO)

It seems like the MySQL service is missing proper authentication details. I've reviewed the workflow file but I'm not entirely sure how to resolve this.

Could anyone help me with configuring MySQL correctly for the CI? Any guidance or suggestions would be greatly appreciated!

**Comment by josdem** [comment]
> Screenshot from 2024-09-29 22-10-19 Hi all,
> 
> I'm encountering a build failure in the CI pipeline related to starting MySQL. The error message is:
> 
> ERROR 1045 (28000): Access denied for user '-p'@'localhost' (using password: NO)
> 
> It seems like the MySQL service is missing proper authentication details. I've reviewed the workflow file but I'm not entirely sure how to resolve this.
> 
> Could anyone help me with configuring MySQL correctly for the CI? Any guidance or suggestions would be greatly appreciated!

Hi @AviraLS22 you have two options to solve this issue:
- Make a branch from this repository instead a fork
- Run tests locally.
